### PR TITLE
VZ-8894 Create the ingress for the Thanos Frontend

### PR DESCRIPTION
**Pull request by davisgomes**
**Changes**
- Create ingress for the query frontend
- Update netpols and authpols to allow correct traffic through the ingresses
- Enable correct service ports for installation
- Add frontend to the VZ status

**Testing**
 - enable thanos component and attempt to access ingress through a browser

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/thanos/thanos.go [hunk 1]
```diff
@@ -40,3 +66,160 @@ func AppendOverrides(_ spi.ComponentContext, _ string, _ string, _ string, kvs [
...
+		IngressName:      constants.ThanosQueryFrontendIngress,
+		HostName:         frontendHostName,
+		TLSSecretName:    frontendCertificateName,
+		ExtraAnnotations: common.SameSiteCookieAnnotations(frontendHostName),
+	}
+	return common.CreateOrUpdateSystemComponentIngress(ctx, thanosProps)
+}
+
+// createOrUpdateComponentAuthPolicy creates the Istio authorization policy for Thanos
+func createOrUpdateComponentAuthPolicy(ctx spi.ComponentContext) error {
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/thanos/thanos.go:117]: If you are modifying the thanos chart it is better to have this in the chart

File: platform-operator/controllers/verrazzano/component/thanos/thanos.go [hunk 2]
```diff
@@ -40,3 +66,160 @@ func AppendOverrides(_ spi.ComponentContext, _ string, _ string, _ string, kvs [
...
+	})
+	if ctrlerrors.ShouldLogKubenetesAPIError(err) {
+		return ctx.Log().ErrorfNewErr("Failed creating/updating Thanos Authorization Policy: %v", err)
+	}
+	return err
+}
+
+// createOrUpdateNetworkPolicies creates or updates network policies for this component
+func createOrUpdateNetworkPolicies(ctx spi.ComponentContext) error {
+	netPolicy := &netv1.NetworkPolicy{ObjectMeta: metav1.ObjectMeta{Name: thanosNetPolicyName, Namespace: ComponentNamespace}}
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/thanos/thanos.go:139]: If you are modifying the thanos chart it is better to have this in the chart

File: platform-operator/controllers/verrazzano/component/thanos/thanos.go (hunk 1 above)
Comment (desagar) [reply platform-operator/controllers/verrazzano/component/thanos/thanos.go:117]: Davis and I discussed putting those in a separate verrazzano sub directory in the templates dir of the thanos chart, if that is ok with you Paul.

File: platform-operator/controllers/verrazzano/vzinstance/verrazzano_instance.go [hunk 3]
```diff
@@ -49,16 +50,18 @@ func GetInstanceInfo(ctx spi.ComponentContext) *v1alpha1.InstanceInfo {
...
+		ElasticURL:             getComponentIngressURL(ingressList.Items, ctx, opensearch.ComponentName, constants.OpensearchIngress),
+		KibanaURL:              getComponentIngressURL(ingressList.Items, ctx, opensearchdashboards.ComponentName, constants.OpensearchDashboardsIngress),
+		GrafanaURL:             getComponentIngressURL(ingressList.Items, ctx, grafana.ComponentName, constants.GrafanaIngress),
+		PrometheusURL:          getComponentIngressURL(ingressList.Items, ctx, promoperator.ComponentName, constants.PrometheusIngress),
+		KialiURL:               getComponentIngressURL(ingressList.Items, ctx, kiali.ComponentName, constants.KialiIngress),
+		JaegerURL:              getComponentIngressURL(ingressList.Items, ctx, jaegeroperator.ComponentName, constants.JaegerIngress),
+		ArgoCDURL:              getComponentIngressURL(ingressList.Items, ctx, argocd.ComponentName, constants.ArgoCDIngress),
+		ThanosQueryFrontendURL: getComponentIngressURL(ingressList.Items, ctx, thanos.ComponentName, constants.ThanosQueryFrontendIngress),
+		ThanosSidecarURL:       getComponentIngressURL(ingressList.Items, ctx, promoperator.ComponentName, constants.ThanosSidecarIngress),
```
Comment (fredtibbitts) [inline platform-operator/controllers/verrazzano/vzinstance/verrazzano_instance.go:64]: I was not thinking we'd expose the sidecar URL. We need it internally but I'm not sure what a user would do with it. Maybe there is value in allowing the user to figure out the sidecar URL, need to give it some more thought.

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**